### PR TITLE
Change uglifier as runtime dependency

### PR DESCRIPTION
**Pull request by bananaappletw**
- [ ] I have added or updated the specs.
- [x] I have verified that the specs pass on my computer.
- [x] I have not attempted to bump, or alter versions.
- [ ] This is a documentation change.
- [x] This is a source change.

### CODE REVIEW

**Comment by envygeeks** [comment]
Uglifier is noteant to be a runtime dependency. It is optional. Our source code hints at this by way of literally seeing if you installed it, not us.

**Comment by bananaappletw** [comment]
@envygeeks Sorry, I don't get it. Isn't it very common to execute `JEKYLL_ENV=production jekyll build` command when using `jekyll`? If so, it's not optional.

**Comment by envygeeks** [comment]
Then there is a bug in the compressor hook. So let me rephrase: It will *not* be forced.

**Comment by bananaappletw** [comment]
@envygeeks Umm, `uglifier` is needed to run the compressor. But the compressor is one of core functionalities. This gem should be installed by default.

**Comment by envygeeks** [comment]
The compressor is, and will remain optional. The logic bug with it being accidentally required without being required has been fixed, this will ***not be changing***.  This is not a "core functionality" this is a "core feature" that is not required, by ***my*** choice as the author of this software.

Thanks!
